### PR TITLE
License on view

### PR DESCRIPTION
**Pull request by lmsilvera**
#332

---
This change is [Reviewable](https://reviewable.io/reviews/opensalt/opensalt/356)

### CODE REVIEW

File: app/Resources/assets/js/cftree/view-documentclass.js [hunk 1]
```diff
@@ -1411,6 +1409,24 @@ function ApxDocument(initializer) {
...
+                let licenseText = licenseDoc[0].licence_text;
+
+                html += '<li class="list-group-item">'
+                + '<strong>License:</strong> '
+                + licenseDoc[0].title;
+
+                if ( licenseText.length > 0) {
+                    html += ' - <i>'
+                    + licenseText
+                    '</i>';
```
Comment (roverwolf) [inline app/Resources/assets/js/cftree/view-documentclass.js]: Looks like a `+` is missing here.

File: src/CftfBundle/Resources/views/DocTree/export.json.twig [hunk 2]
```diff
@@ -45,9 +45,12 @@
     ,"CFConcepts": [
     ]
     {%- endif %}
-    {% if licences is not empty -%}
-    {#- TODO: CFLicenses -#}
+    {% if licence is not empty -%}
     ,"CFLicenses": [
+        {
+            "title": {{ licence.title |json_encode |raw }}
+            , "licence_text": {{ licence.licenceText |json_encode |raw }}
```
Comment (roverwolf) [inline src/CftfBundle/Resources/views/DocTree/export.json.twig]: Should be `licenseText`, and are missing the other attributes required here: Ref: [IMS CASE Spec - CFLicense](https://www.imsglobal.org/sites/default/files/CASE/casev1p0/information_model/caseservicev1p0_infomodelv1p0.html#Interface_CFLicense)

File: src/CftfBundle/Controller/DocTreeController.php [hunk 3]
```diff
@@ -210,7 +210,7 @@ public function exportAction(Request $request, LsDoc $lsDoc)
             'itemTypes' => $itemTypes,
             'subjects' => $lsDoc->getSubjects(),
             'concepts' => [],
-            'licences' => [],
+            'licence' => $lsDoc->getLicence(),
```
Comment (roverwolf) [inline src/CftfBundle/Controller/DocTreeController.php]: Per the spec, it is possible for each item to have a different licence than the document, hence the array that was here.

File: src/CftfBundle/Resources/views/DocTree/export.json.twig [hunk 4]
```diff
@@ -43,11 +43,20 @@
...
-    {#- TODO: CFLicenses -#}
     ,"CFLicenses": [
+    {% for licence in licences %}
+    {% if not loop.first %},{% endif %}
+        {
+            "identifier": {{ licence.identifier |json_encode |raw }}
+            , "uri": {{ licence.uri |json_encode |raw }}
+            , "title": {{ licence.title |json_encode |raw }}
+            , "description": {{ licence.description |json_encode |raw }}
+            , "licenceText": {{ licence.licenceText |json_encode |raw }}
```
Comment (roverwolf) [inline src/CftfBundle/Resources/views/DocTree/export.json.twig]: The name of the field in the JSON should be `licenseText` as defined in the spec.

File: app/Resources/assets/js/cftree/view-documentclass.js [hunk 5]
```diff
@@ -1411,6 +1409,24 @@ function ApxDocument(initializer) {
...
             }
+
+            // render license information in case of framework has one.
+            if ("undefined" !== typeof self.licenses && self.licenses.length > 0) {
+                let licenseDoc = self.licenses;
+                let licenseText = licenseDoc[0].licenseText;
+
+                html += '<li class="list-group-item">'
+                + '<strong>License:</strong> '
+                + licenseDoc[0].title;
```
Comment (roverwolf) [inline app/Resources/assets/js/cftree/view-documentclass.js]: We should probably wrap the display of the title with `render.escaped()` for safety.

File: app/Resources/assets/js/cftree/view-documentclass.js [hunk 6]
```diff
@@ -1411,6 +1409,24 @@ function ApxDocument(initializer) {
...
+
+                if ( licenseText.length > 0) {
+                    html += ' - <i>'
+                    + licenseText
```
Comment (roverwolf) [inline app/Resources/assets/js/cftree/view-documentclass.js]: We should also wrap the display of the text with `render.escaped()` as well.

**Comment by lmsilvera** [comment]
@roverwolf PR ready to be merged

**Comment by roverwolf** [comment]
@lmsilvera You need to update it with the latest from develop before we can merge it.

**Comment by lmsilvera** [comment]
@roverwolf PR already updated 👍

**Pull request merged**